lib/pf/Switch/Ubiquiti/Unifi.pm

```diff
@@ -269,16 +269,25 @@ sub _deauthenticateMacWithHTTP {
         }
     }
 
-    if ($found) {
-        $response = $ua->post("$base_url/api/s/$site_opts{'name'}/cmd/stamgr", Content => encode_json($args));
-        if ($response->is_success) {
-            $logger->info("Deauth on site: $site_opts{'desc'}");
-        }
-    } else {
-        foreach my $entry (@{$json_data->{'data'}}) {
-            $response = $ua->post("$base_url/api/s/$entry->{'name'}/cmd/stamgr", Content => encode_json($args));
-            if ($response->is_success) {
-                $logger->info("Deauth on site: $entry->{'desc'}");
+    tie my %SwitchConfig, 'pfconfig::cached_hash', "config::Switch($host_id)";
+    foreach my $switch_id (keys(%SwitchConfig)) {
+        my $switch = $SwitchConfig{$switch_id};
+        # If the other switch is a MAC based entry and an Ubiquiti AP and is part of the same AP group, then we'll send the deauth
+        if(valid_mac($switch_id) && $switch->{type} eq $self->{_type} && $switch->{group} eq $self->{_group}) {
+            $logger->info("Performing deauth for AP $switch_id");
+            $args->{ap_mac} = $switch_id;
+            if ($found) {
+                $response = $ua->post("$base_url/api/s/$site_opts{'name'}/cmd/stamgr", Content => encode_json($args));
+                if ($response->is_success) {
+                    $logger->info("Deauth on site: $site_opts{'desc'}");
+                }
+            } else {
+                foreach my $entry (@{$json_data->{'data'}}) {
+                    $response = $ua->post("$base_url/api/s/$entry->{'name'}/cmd/stamgr", Content => encode_json($args));
+                    if ($response->is_success) {
+                        $logger->info("Deauth on site: $entry->{'desc'}");
```

Comment:
Can we break the loop here ?

Reply:
The goal is to deauth on all the APs of the switch group, wouldn't breaking here stop it from happening ?

Reply:
Yes i know but if it succeed , does it mean i found the mac on the ap ?
Or it succeed every time ?

Reply:
From what I was told, it will succeed even if the MAC is not on the AP but I still need to test this in the field with my customer to confirm this 100%

Reply:
Ok , so don't break the loop.
Btw we will have a lot of logs.

Reply:
There should be 1 line for each AP we do the deauth for. We could always get this in debug but I wanted to try to make it explicit it was doing it on multiple APs